src/mydumper_working_thread.c

```diff
@@ -1152,7 +1157,7 @@ gboolean new_db_table( struct db_table **d, MYSQL *conn, struct configuration *c
     dbt = g_new(struct db_table, 1);
     dbt->status = UNDEFINED;
     dbt->database = database;
-    dbt->table = g_strdup(table);
+    dbt->table = backtick_protect(table);
```

Comment:
It passed quite long test with mischievous test cases adapted from mysqldump. See above answer and bug description.

Reply:
[mydumper.test](https://gist.github.com/midenok/63be552ca99b7bd844d7a28b657a7cfc#file-mydumper-test)

Reply:
can you share the include files so I can add this test to the project?

Reply:
* [mydumper.test](https://gist.github.com/midenok/63be552ca99b7bd844d7a28b657a7cfc#file-mydumper-test)
* [mydumper.result](https://gist.github.com/midenok/63be552ca99b7bd844d7a28b657a7cfc#file-mydumper-result)
* [mydumper_dump.inc](https://gist.github.com/midenok/63be552ca99b7bd844d7a28b657a7cfc#file-mydumper_dump-inc)
* [mydumper_restore.inc](https://gist.github.com/midenok/63be552ca99b7bd844d7a28b657a7cfc#file-myloader_restore-inc)

These are run under [MTR](https://dev.mysql.com/doc/dev/mysql-server/latest/PAGE_MYSQLTEST_LANGUAGE_REFERENCE.html).

I'm not sure how would you run it because it checks the output against the .result file, but at least you can check runnability.

Reply:
you should add `-M` to mydumper_args on mydumper_dump.inc, to check that the consistency